### PR TITLE
Loops in process and scenario orchestration

### PR DESCRIPTION
**Pull request by Nemikor**
# Description

Add doWhile and for loop in process and scenario orchestration DSL.

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing (regression) unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public abstract sealed class ForScenarioProvidersBaseDefinition<
 
   /**
    * Attaches and changes scope to a conditional branch that is only executed if the given <code>
-   * predicate</code> matches at runtime for an integration call of the underlying scenario.
+   * iterations</code> matches at runtime for an integration call of the underlying scenario.
```
Comment (MartinBuchheim) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java]: revert to previous wording

File: sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java [hunk 2]
```diff
@@ -56,6 +57,24 @@ public abstract sealed class ForScenarioProvidersBaseDefinition<
     return def.elseIfCase(predicate);
   }
 
+  public WhileLoopCallScenarioConsumerDefinition<S, M>.Branch<
```
Comment (MartinBuchheim) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java:72]: javadoc

File: sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java [hunk 3]
```diff
@@ -56,6 +57,24 @@ public abstract sealed class ForScenarioProvidersBaseDefinition<
...
+          WhileLoopCallScenarioConsumerDefinition<S, M>>
+      doWhile(final Predicate<ScenarioOrchestrationContext<M>> predicate) {
+    final var def =
+        new WhileLoopCallScenarioConsumerDefinition<S, M>(self(), getIntegrationScenario());
+    nodes.add(def);
+    return def.doWhile(predicate);
+  }
+
+  public ForLoopCallScenarioConsumerDefinition<S, M>.Branch<
```
Comment (MartinBuchheim) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java:93]: javadoc

**Review by MartinBuchheim — CHANGES_REQUESTED** [review]

**Comment by MartinBuchheim** [comment]
Please also rename endLoop() to endDoWhile() for while-loops

File: sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java (hunk 1 above)
Comment (Nemikor) [reply sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/orchestration/scenario/dsl/ForScenarioProvidersBaseDefinition.java]: reverted

**Review by MartinBuchheim — DISMISSED** [review]

**Review by malte-westendorf-ikor — DISMISSED** [review]

**Review by MartinBuchheim — APPROVED** [review]

**Pull request merged**